### ISSUE TITLE
Make contextualIdentities an optional permission.

### ISSUE THREAD

**Issue by ghost** [mrdokenny/Cookie-AutoDelete#152]
Cookie AutoDelete changing this internal Firefox preference:
`privacy.userContext.enabled;true`
in about:config without my permission and without my knowledge.

My Default preference is:
`privacy.userContext.enabled;false`

I also don't have ability to control (enable/disable) this "feature".

Enabling or disabling "Enable Support for Firefox's Contextual Identities (EXPERIMENTAL Firefox 53+)" setting in Cookie AutoDelete option doesn't change anything.

I'm using Cookie AutoDelete 1.4.4 + Firefox Nightly 57.0a1.

Please provide ability to control (enable/disable) this "feature" and some information after installing this extension.

**Comment by drdaeman** [comment]
Are you sure it's C-AD that does and not e.g. Mozilla rolling out the Contextual Identities in Nightly?

As far as I know, WebExtensions [have no way](https://wiki.mozilla.org/WebExtensions/FAQ#Will_I_have_access_to_about:config_or_the_preferences.3F) to access `about:config` settings. Neither a string `userContext` appears anywhere in the source code of C-AD.

**Comment by ke-d** [comment]
In the 1.X.X versions, there's a check to see if the `contextualIdentities` API exists (since Chromium browsers do not have this feature as well as older versions of Firefox). I think just "checking" is enough to activate the preference.

In 2.X.X, I'm removing that check and only using browser detection but the hard minimum to use C-AD will be 56+ as there are some problems with < 55 #41

**Comment by ke-d** [comment]
![screenshot from 2017-09-26 08-10-45](https://user-images.githubusercontent.com/17077613/30867957-385996e6-a292-11e7-948e-8cc9c0a1cb60.png)

This is interesting. I think sometime in the future I'll make `contextualIdentities` an optional permission.

**Comment by jonathanKingston** [comment]
@mrdokenny `contextualIdentities` permission now does this, we/I will be working on the optional permission change soon (currently this isn't possible).

See: https://hacks.mozilla.org/2017/10/containers-for-add-on-developers/ on the changes we made for 57. This links to bugs that we will be working on next.

**Comment by ke-d** [comment]
@jonathanKingston I was actually under the impression that it was already implemented. I guess not. 

Thanks for the heads up though.

**Comment by jonathanKingston** [comment]
Nope sorry individual APIs need their own support for optional API support.

The bug is: https://bugzilla.mozilla.org/show_bug.cgi?id=1386673 if you want to follow it.

**Comment by maxoku** [comment]
Any progress in the matter? I would really like to disable the container tabs.

**Comment by kennethtran93** [comment]
Follow the bugzilla link above for progress as it is a browser limitation at this time.

TL;DR at this time, not yet.

Closing for now due to browser limitations.  Feel free to comment on this to reopen it or create a new issue (when `contextualIdentities` can be made optional).
